jupyterhub/handlers/pages.py

```diff
@@ -591,6 +606,69 @@ def get(self, *args):
         self.finish()
 
 
+class Pagination(BaseHandler):
+
+    _page_name = 'page'
+    _per_page_name = 'per_page'
+    _default_page = 1
+    _default_per_page = 100
+    _max_per_page = 250
+    _record_name = 'users'
+    _display_msg = (
+        'Displaying {record_name} <b>{start} - {end}</b>. Total {record_name}: {total}'
+    )
+
+    def __init__(self, *args, **kwargs):
+        """Potential parameters.
+        **url**: URL in request
+        **page**: current page in use
+        **per_page**: number of records to display in the page. By default 100
+        **total**: total records considered while paginating
+        **display_msg**: informative text for pagination
+        **record_name**: name of the record, showed in pagination info
+        """
+        self.page = kwargs.get(self._page_name, 1)
```

Comment:
I don't think this needs to be so customizable as to have the name of the argument used to get the page as a class attribute.

I think defining args in init is probably best.

Reply:
In this case, I would prefer to maintain these in this way to don't copy the strings across methods or to allow to use the `Pagination.get_page_args(self)` as it's currently coded and without needing to fully instantiate the class to get the args to later re-instantiate the class again.